### PR TITLE
docs: Make all examples in README full examples

### PR DESCRIPTION
**Pull request by matthewfeickert**
# Description

Following up on feedback by @dantrim from a long time ago (I can't find the original communication?), this PR makes all the examples in the `README` full examples with generating code snippets. It additionally updates the images used in the `README`:

* 1 bin example:
README_1bin_example

* 2 bin example:
README_2bin_example

ReadTheDocs build: https://pyhf.readthedocs.io/en/docs-make-examples-in-readme-examples

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add code to all examples in README
* Add new images to README and remove old ones
   - Rename images to have clearer names
   - Rename ROOT 2-bin example to match hfh prefix notation
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
One note: running

```python
import pyhf
import numpy as np
import matplotlib.pyplot as plt
import pyhf.contrib.viz.brazil

pyhf.set_backend("numpy")
model = pyhf.simplemodels.hepdata_like(
   signal_data=[10.0], bkg_data=[50.0], bkg_uncerts=[7.0]
)
data = [55.0] + model.config.auxdata

results = []
poi_vals = np.linspace(0, 5, 41)
for test_poi in poi_vals:
    results.append(
        pyhf.infer.hypotest(
            test_poi, data, model, qtilde=True, return_expected_set=True
        )
    )

fig, ax = plt.subplots()
fig.set_size_inches(7, 5)
ax.set_xlabel(r"$\mu$ (POI)")
ax.set_ylabel(r"$\mathrm{CL}_{s}$")
pyhf.contrib.viz.brazil.plot_results(ax, poi_vals, results)
fig.savefig("README_1bin_example.png")
```
does generate the warning
```
/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/calculators.py:186: RuntimeWarning: invalid value encountered in double_scalars
  teststat = (qmu - qmu_A) / (2 * self.sqrtqmuA_v)
```

**Comment by matthewfeickert** [comment]
@kratsg @lukasheinrich To review this PR I would encourage you to check it out locally and then do a search and replace for

```
master/docs/_static/img
```
to
```
docs/make-examples-in-README-examples/docs/_static/img
```
in `README.rst` and then build the docs locally. Otherwise I think it's a bit confusing to get the changes in the images from the diffs.

**Comment by matthewfeickert** [comment]
@dantrim your review here would be welcomed as well.

File: README.rst [hunk 1]
```diff
@@ -84,13 +84,41 @@ A one bin example
...
+   pyhf.set_backend("numpy")
+   model = pyhf.simplemodels.hepdata_like(
+       signal_data=[10.0], bkg_data=[50.0], bkg_uncerts=[7.0]
+   )
+   data = [55.0] + model.config.auxdata
+
+   results = []
+   poi_vals = np.linspace(0, 5, 41)
+   for test_poi in poi_vals:
+       results.append(
```
Comment (lukasheinrich) [inline README.rst:101]: could make this a list comprehension

File: README.rst [hunk 2]
```diff
@@ -101,15 +129,42 @@ A two bin example
...
+   model = pyhf.simplemodels.hepdata_like(
+       signal_data=[30.0, 45.0], bkg_data=[100.0, 150.0], bkg_uncerts=[15.0, 20.0]
+   )
+   data = [100.0, 145.0] + model.config.auxdata
+
+   results = []
+   poi_vals = np.linspace(0, 5, 41)
+   for test_poi in poi_vals:
+       results.append(
+           pyhf.infer.hypotest(
```
Comment (lukasheinrich) [inline README.rst:147]: see above

File: README.rst (hunk 1 above)
Comment (matthewfeickert) [reply README.rst:101]: Yeah, good point. I'll fix that.

**Comment by matthewfeickert** [comment]
> does generate the warning
> 
> ```
> /home/feickert/Code/GitHub/pyhf/src/pyhf/infer/calculators.py:186: RuntimeWarning: invalid value encountered in double_scalars
>   teststat = (qmu - qmu_A) / (2 * self.sqrtqmuA_v)
> ```

This is only because we start the scan at `mu=0`

```python
poi_vals = np.linspace(0, 5, 41)
```

and so isn't really an issue.

**Pull request merged**